src/app/navbar/navbar.component.ts

```diff
@@ -63,6 +63,10 @@ export class NavbarComponent extends Logout implements OnInit {
   ngOnInit() {
     this.userQuery.user$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((user) => {
       this.user = user;
+      if (this.user) {
+        localStorage.setItem(acceptedTOSVersion, user.tosversion);
```

Comment:
I have concerns about using localStorage for this:

1. The same user needs to accept the TOS on every computer the user logs in to Dockstore
2. This could "carry over" between users, although I didn't double-check the flow.
    1. User A accepts TOS, logs out
    2. Use B logs on same computer (unlikely, but shared computer, multiple accounts for same person) and localStorage indicates TOS are accepted, even if B hadn't accepted.

Reply:
Okay, I don't think 1 is that big a of deal, but yes there would be carry over if people shared the same computer/browser window. Maybe instead of logging them out, I'll just display the banner that follows the user around with somewhat different wording.

Reply:
> Okay, I don't think 1 is that big a of deal, but yes there would be carry over if people shared the same computer/browser window. Maybe instead of logging them out, I'll just display the banner that follows the user around with somewhat different wording.

I recognize this makes it more complicated, alas. You could go with the annoying banner, or you'd have to introduce a new page after they log in if they haven't accepted the latest TOS.